src/modules/doctors/doctors.service.ts

```diff
@@ -11,27 +12,34 @@ export class DoctorsService {
   constructor(
     @InjectRepository(Doctor)
     private doctorsRepository: Repository<Doctor>,
-    @InjectRepository(User)
-    private usersRepository: Repository<User>,
-    @InjectRepository(Specialization)
-    private specializationsRepository: Repository<Specialization>,
+    private readonly usersService: UsersService,
+    private readonly specializationsService: SpecializationsService,
   ) {}
   async create(createDoctorInput: CreateDoctorInput): Promise<Doctor> {
-    const doctor = await this.doctorsRepository.create(createDoctorInput);
-    doctor.specialization = await this.specializationsRepository.findOne({
-      id: createDoctorInput.specializationId,
-    });
-    doctor.user = await this.usersRepository.findOne({
-      id: createDoctorInput.userId,
-    });
-
     try {
```

Comment:
You don't need a try/catch here, as it has no context reverse logic

Reply:
I left this try-catch block for now to correctly log errors, I will refactor that in one of the next MRs.